portfolio/src/main/webapp/index.html

```diff
@@ -6,14 +6,19 @@
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
   </head>
-  <body>
+  <body onload="testFetch()">
```

Comment:
Instead of setting up handlers in your html -- could you do so via JS? (Look for "DomContentLoaded")